### PR TITLE
Grant `permission.POST_NOTIFICATIONS` in Macrobenchmark tests

### PR DESCRIPTION
**Pull request by astamato**
Description of the problem:
Benchmarks failing because this dialog is in the middle.

aa

Expected solution:

```
@get:Rule
val grantPermissionRule = GrantPermissionRule.grant(Manifest.permission.POST_NOTIFICATIONS)

@get:Rule
val benchmarkRule = MacrobenchmarkRule()
```

This [issue](https://issuetracker.google.com/283745750) is tracking the alternatives.
Actual solution: manually dismissing the Alert Dialog or executing the command programatically.

### CODE REVIEW

File: benchmarks/src/main/java/com/google/samples/apps/nowinandroid/foryou/ForYouActions.kt [hunk 1]
```diff
@@ -22,6 +22,11 @@ import androidx.test.uiautomator.Until
 import androidx.test.uiautomator.untilHasChildren
 import com.google.samples.apps.nowinandroid.flingElementDownUp
 
+fun MacrobenchmarkScope.allowNotifications() {
+    val dialog = device.wait(Until.findObject(By.text("Allow")), 5_000)
+    dialog?.click()
```
Comment (astamato) [inline benchmarks/src/main/java/com/google/samples/apps/nowinandroid/foryou/ForYouActions.kt:27]: Am I missing a device.waitForIdle() ?

Comment (SimonMarquis) [reply benchmarks/src/main/java/com/google/samples/apps/nowinandroid/foryou/ForYouActions.kt:27]: Tested on an emulator with API 33 and this works just fine 👍

**Comment by dturner** [comment]
@astamato Do you want to change this PR to "Ready for review" or are you still working on it?

**Comment by astamato** [comment]
@dturner Not ready yet, TY

File: benchmarks/src/main/java/com/google/samples/apps/nowinandroid/GeneralActions.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+     device.wait(Until.gone(obj), 5_000)
+     }
+     *
+     * or (preferred) executing the grant command on the target package.
+     */
+    if (SDK_INT >= TIRAMISU) {
+        val command = "pm grant $packageName ${permission.POST_NOTIFICATIONS}"
+        device.executeShellCommand(command)
+    }
+}
```
Comment (SimonMarquis) [inline benchmarks/src/main/java/com/google/samples/apps/nowinandroid/GeneralActions.kt:42]: Let's make it an actual javadoc comment (with code highlighting) 😁

```suggestion
/**
 * Because the app under test is different from the one running the instrumentation test,
 * the permission has to be granted manually by either
 * 
 * - tapping the Allow button
 *    ```kotlin
 *    val obj = By.text("Allow")
 *    val dialog = device.wait(Until.findObject(obj), TIMEOUT)
 *    dialog?.let {
 *        it.click()
 *        device.wait(Until.gone(obj), 5_000)
 *    }
 *    ```
 * - or (preferred) executing the grant command on the target package.
 */
fun MacrobenchmarkScope.allowNotifications() {
    if (SDK_INT >= TIRAMISU) {
        val command = "pm grant $packageName ${permission.POST_NOTIFICATIONS}"
        device.executeShellCommand(command)
    }
}
```

Comment (SimonMarquis) [reply benchmarks/src/main/java/com/google/samples/apps/nowinandroid/GeneralActions.kt:42]: image

Comment (astamato) [reply benchmarks/src/main/java/com/google/samples/apps/nowinandroid/GeneralActions.kt:42]: Ah great idea, thanks :)

File: benchmarks/src/main/java/com/google/samples/apps/nowinandroid/GeneralActions.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+import android.Manifest.permission
+import android.os.Build.VERSION.SDK_INT
+import android.os.Build.VERSION_CODES.TIRAMISU
+import androidx.benchmark.macro.MacrobenchmarkScope
+
+/**
+ * Because the app under test is different from the one running the instrumentation test,
+ * the permission has to be granted manually by either:
+ *
```
Comment (SimonMarquis) [inline benchmarks/src/main/java/com/google/samples/apps/nowinandroid/GeneralActions.kt:27]: Sorry for the extraneous space 🙈

```suggestion
 *
```

File: benchmarks/src/main/java/com/google/samples/apps/nowinandroid/interests/TopicsScreenScrollBenchmark.kt
Comment (SimonMarquis) [inline benchmarks/src/main/java/com/google/samples/apps/nowinandroid/interests/TopicsScreenScrollBenchmark.kt:1]: Is this file intended to be part of this PR?

Comment (astamato) [reply benchmarks/src/main/java/com/google/samples/apps/nowinandroid/interests/TopicsScreenScrollBenchmark.kt:1]: Yes. I can split into another PR if its confusing.

**Review by dturner — APPROVED** [review]
